### ISSUE TITLE
No file named error

### ISSUE THREAD

**Issue by KamilLelonek** [KronicDeth/intellij-elixir#203]
I configured Intellij with Elixir plugin as described in `README.md`.

I'm using OS X and brew-installed Elixir (`/usr/local/Cellar/elixir/1.1.1/bin/elixir`).

When I run test configuration I get:

![image](https://cloud.githubusercontent.com/assets/3199333/11016461/9a5db6be-853f-11e5-8942-000a77d0d037.png)

Everything is done as described in manual.

**Comment by KronicDeth** [comment]
@zyuyou, any ideas?

**Comment by zyuyou** [comment]
@KamilLelonek can you show the run configuration?

**Comment by davidcotter** [comment]
I have this issue too if you would like any info

**Comment by zyuyou** [comment]
@KamilLelonek @davidcotter the Elixir plugin only support `Mix` runconfigure which based on the `Elixir`'s mix executable tool by far. so  running test configuration should ensure that the project is a `mix-project` which can be created by command `mix new [app_name]`

I will try to make it support `ExUnit` if I have free time to work for it.

**Comment by KamilLelonek** [comment]
The issue happened during "Import project from external model". I did it once again, selecting `external model` instead of `existing sources` and it works.

**Comment by 8822** [comment]
I'm getting the same error.  Any workarounds?

**Comment by davidcotter** [comment]
Hi

I think I am doing something wrong. I have never been able to run a project from IntelliJ and always get this same error 

"C:\Program Files (x86)\Elixir\bin\elixir.bat" ""
"No file named "

Maybe it doesn't work on Windows? I have trued it with mix - with the latest Intellij - the latest elixir plugin - with a new project - with an existing project - I think I am missing something basic.

Thanks

**Comment by KronicDeth** [comment]
The argument to `elixir.bat` is an empty string, `""`.  It means your project is missing the path to whatever it is trying to run with `elixir.bat`.  Have you tried [@KamilLelonek's process](https://github.com/KronicDeth/intellij-elixir/issues/203#issuecomment-155842779)?

It works on Windows: whenever someone reports a Windows bug, I debug and make test builds from Windows.  I have a Windows box for gaming primarily and an OSX laptop for primary development.  The only environment I don't test in (unless I got around to setting up an environment in a VM) is Linux.

**Comment by zyuyou** [comment]
![2016-03-22 10 41 16](https://cloud.githubusercontent.com/assets/1913492/13940725/0da41b04-f01b-11e5-9fb1-db4191850677.png)

@davidcotter Is it possible that you forgot to input a `Command` ?

**Comment by davidcotter** [comment]
Thanks.

If I use the "Import project from external model" it uses mix ad the
command (not elixir) and you can pas in "compile" or any other mix command
and it runs. Thanks.

**Comment by KronicDeth** [comment]
@zyuyou users keep having this issue when not importing from mix.  Do you think we should should detect if they are using mix when they pick import project without select import from mix?  When does the non-mix import work?

**Comment by Nihisil** [comment]
I think it should be some kind of auto detection of mix project structure and suggesting to import all needed files.

After installation of Elixir plugin I have created new project inside Intellij IDEA and plugin wasn't work with it properly. I had same "file not found" bug. 

Only after looking into this issue on github I was able to create mix project from console and import it. It resolved empty-file issue.

**Comment by KronicDeth** [comment]
I agree with @Nihisil that it would probably do auto-detect.  @zyuyou based the project support on intellij-erlang's where rebar support is equivalent to mix support, but unlike rebar in Erlang project, mix is more prevalent in Elixir projects, so assuming mix support is probably not a bad idea for intellij-elixir.  We'll get to it eventually.  For now, there is a work-around, unlike a lot of other features, so it will be lower priority since people only have to remember how to import a project once per project.

**Comment by zyuyou** [comment]
There is an difference between `rebar` and `mix`, `mix` do nothing without `mix.exs` but `rebar` can.
<img width="599" alt="2016-06-05 12 52 24" src="https://cloud.githubusercontent.com/assets/1913492/15800872/3b58745a-2ab8-11e6-81b1-a863f120da76.png">

You means giving some tips when they run `mix` run configuration  in a non-mix project?

**Comment by KronicDeth** [comment]
@zyuyou I was thinking that the user should not have to do `File > New > Project from existing Sources > Import from external model > Mix`, but instead doing `File > Open` and picking a directory with a `mix.exs` will assume it's a mix project or prompt to set it up as one.  Users seem to keep missing that they need to import using the MIx external model to get the Run configurations to work.  I'm not sure how we would go about doing that detection and prompting however.  Rubymine seems to do that for detecting Rails projects, but I'm not sure how it does that.

**Comment by zyuyou** [comment]
@KronicDeth [MixProjectOpenProcessor](https://github.com/KronicDeth/intellij-elixir/blob/master/src/org/elixir_lang/mix/importWizard/MixProjectOpenProcessor.java) can do this work with `<projectOpenProcessor implementation="org.elixir_lang.mix.importWizard.MixProjectOpenProcessor" />`, and it works with mix project, but prompt to set it up as one, because we don't know if it will work fine after set it up as a mix project.

**Comment by HangingClowns** [comment]
So what is the solution? i don't understand?

**Comment by fhsgoncalves** [comment]
@HangingClowns the solution is import the project using the option `import as a mix project`, like that:

<img width="708" alt="screen shot 2017-06-15 at 13 50 37" src="https://user-images.githubusercontent.com/8892624/27192429-b1ff7108-51d1-11e7-9eff-1857690629a3.png">

It worked for me :)

**Comment by HangingClowns** [comment]
@fhsgoncalves correct! :)

**Comment by sheshankkodam** [comment]
@fhsgoncalves Thanks. Importing as Mix project and setting the run configurations as shown below worked. 
![image](https://user-images.githubusercontent.com/15146327/42048508-b8077190-7ac0-11e8-9e5b-f3c9111a84fe.png)
